pkg/rpc/server/server.go

```diff
@@ -534,7 +545,15 @@ func (s *Server) getUTXOTransfers(ps request.Params) (interface{}, *response.Err
 	}
 	tr := new(state.Transfer)
 	err = s.chain.ForEachTransfer(addr, tr, func() error {
-		if tr.Timestamp < start || end != 0 && tr.Timestamp > end {
+		var count int
+		for _, res := range sent {
+			count += len(res.Transactions)
+		}
+		for _, res := range recv {
+			count += len(res.Transactions)
+		}
+		if tr.Timestamp < start || end != 0 && tr.Timestamp > end ||
+			(limit != 0 && count >= limit) {
```

Comment:
Shouldn't we include the `limit`? I.e. `count>limit`, as far as if we want to get the maximum number of transfers = 1, we'll get nothing.

Reply:
No, it works as intended because `count == 1` exactly when there is one element already in the output, so we can stop iterating. Technically, we can stop it earlier (at the end of this function), but this approach works fine too.